nucliadb_node/tests/test_search_fields_filtering.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright (C) 2021 Bosutech XXI S.L.
+//
+// nucliadb is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at info@nuclia.com.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+
+mod common;
+
+use ::nucliadb_protos::{nodereader, noderesources, nodewriter};
+use common::NodeFixture;
+use nucliadb_core::protos::{self as nucliadb_protos, NewShardRequest};
+use nucliadb_protos::prost_types::Timestamp;
+use std::collections::HashMap;
+use std::time::SystemTime;
+use tonic::Request;
+use uuid::Uuid;
+
+const VECTOR_DIMENSION: usize = 10;
+
+#[tokio::test]
+async fn test_search_fields_filtering() -> Result<(), Box<dyn std::error::Error>> {
+    let mut fixture = NodeFixture::new();
+    fixture.with_writer().await?.with_reader().await?;
+    let mut writer = fixture.writer_client();
+    let mut reader = fixture.reader_client();
+
+    let new_shard_response = writer.new_shard(Request::new(NewShardRequest::default())).await?;
+    let shard_id = &new_shard_response.get_ref().id;
+
+    // Add a resource with vectors
+    let resource = build_resource_with_field(shard_id.clone(), "f/field1".to_string());
+    let result = writer.set_resource(resource).await?;
+
+    assert_eq!(result.get_ref().status(), nodewriter::op_status::Status::Ok);
+
+    // Search filtering with an unexisting field, to check that no vector are returned
+    let magnitude = f32::sqrt((17.0_f32).powi(2) * VECTOR_DIMENSION as f32);
+    let query_vector = vec![17.0 / magnitude; VECTOR_DIMENSION];
+
+    let search_request = nodereader::SearchRequest {
+        shard: shard_id.clone(),
+        fields: ["f/foobar".to_string()].to_vec(),
+        vector: query_vector,
+        page_number: 0,
+        result_per_page: 1,
+        min_score_semantic: -1.0,
+        paragraph: false,
+        document: false,
+        ..Default::default()
+    };
+    let results = reader.search(Request::new(search_request)).await?.into_inner();
+
+    assert!(results.vector.is_some());
+    let vector_results = results.vector.unwrap();
+    assert_eq!(vector_results.documents.len(), 0);
+
+    // Search filtering with a real field, to check that the vector is returned
+    let query_vector2 = vec![17.0 / magnitude; VECTOR_DIMENSION];
```

Comment:
Why aren't you using `query` again? You could `.clone()` it in the previous call if you needed to